java/src/jmri/jmrit/display/layoutEditor/LevelXing.java

```diff
@@ -633,28 +632,28 @@ public LayoutTrack getConnectD() {
 
     public void setConnectA(LayoutTrack o, int type) {
         connectA = o;
-        if ((connectA != null) && (type != TRACK)) {
+        if ((connectA == null) ? (type != NONE) : ((!(o instanceof TrackSegment)) || (type != TRACK))) {
```

Comment:
https://guide.freecodecamp.org/c/ternary-operator/

Reply:
`      if (connectA == null) {
            if (type != NONE) {
                log.error("unexpected type of A connection to levelXing - " + type);
            }
        } else {
            if ((!(o instanceof TrackSegment)) || (type != TRACK)) {
                log.error("unexpected type of A connection to levelXing - " + type);
            }
        }
`
So 10 lines of code instead of 3… and DRY fail.